tealeaf/platform/TeaLeafViewController.mm

```diff
@@ -255,12 +255,22 @@ - (void) destroyGLView {
 - (void) createGLView {
 	//create our openglview and size it correctly
 	OpenGLView *glView = [[OpenGLView alloc] initWithFrame:self.appDelegate.initFrame];
+	[self.view addSubview:glView];
     
-	self.view = glView;
 	self.appDelegate.canvas = glView;
     
 	core_init_gl(1);
     
+	glView.translatesAutoresizingMaskIntoConstraints = NO;
+    
+	if (@available(iOS 11.0, *)) {
+		UILayoutGuide * layoutGuide = self.view.safeAreaLayoutGuide;
+		[glView.leadingAnchor constraintEqualToAnchor:layoutGuide.leadingAnchor].active = YES;
+		[glView.trailingAnchor constraintEqualToAnchor:layoutGuide.trailingAnchor].active = YES;
+		[glView.topAnchor constraintEqualToAnchor:layoutGuide.topAnchor].active = YES;
+		[glView.bottomAnchor constraintEqualToAnchor:layoutGuide.bottomAnchor].active = YES;
+	}
```

Comment:
will this work for iphone x ?

Reply:
yes @rampr, this will work.

Reply:
can we remove the old change we made for iphone x then ?

Reply:
I was thinking about it, will do the same

Reply:
old change also removed.